packages/xod-client-electron/src/arduinoDependencies/messages.js

```diff
@@ -1,12 +1,38 @@
+const librariesMissing = libraryNames =>
+  libraryNames.length
+    ? `You have to install these libraries first: ${libraryNames} \r\n`
+    : '';
+const librariesInstalled = libraryNames =>
+  libraryNames.length ? `Libraries ${libraryNames} installed \r\n` : '';
+
+const packagesMissing = packageNames =>
+  packageNames.length
+    ? `You have to install these packages first: ${packageNames}`
+    : '';
+const packagesInstalled = packageNames =>
+  packageNames.length ? `Packages ${packageNames} installed` : '';
+
 export default {
-  ARDUINO_LIBRARIES_MISSING: ({ libraries, libraryNames }) => ({
-    title: 'Some arduino libraries are missing',
-    solution: `You have to install these libraries first: ${libraryNames}`,
+  ARDUINO_DEPENDENCIES_MISSING: ({
+    libraries,
+    libraryNames,
+    packages,
+    packageNames,
+  }) => ({
+    title: 'Some arduino dependencies are missing',
+    solution: [
+      librariesMissing(libraryNames),
+      packagesMissing(packageNames),
```

Comment:
Extracting a function in `messages.js` hurts more than helps. Suggest embedding.